daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -396,28 +366,23 @@ private[lf] object SBuiltin {
   // Text functions
   //
   final case object SBExplodeText extends SBuiltinPure(1) {
-    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue =
+    override private[speedy] def executePure(args: util.ArrayList[SValue]): SList =
       SList(FrontStack(Utf8.explode(getSText(args, 0)).map(SText)))
   }
 
   final case object SBImplodeText extends SBuiltinPure(1) {
-    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue = {
-      args.get(0) match {
-        case SList(xs) =>
-          val ts = xs.map {
-            case SText(t) => t
-            case v =>
-              throw SErrorCrash(s"type mismatch implodeText: expected SText, got $v")
-          }
-          SText(Utf8.implode(ts.toImmArray))
-        case _ =>
-          throw SErrorCrash(s"type mismatch implodeText: $args")
+    override private[speedy] def executePure(args: util.ArrayList[SValue]): SText = {
+      val xs = getSList(args, 0)
+      val ts = xs.map {
```

Comment:
any reason not to use `getSText` here?

Reply:
The signature of `getSText` is not adapted. Maybe will do it in later.